### ISSUE TITLE
dbplyr 2.4.0 compatibility issues with bigrquery 1.4.2

### ISSUE THREAD

**Issue by aarizvi** [tidyverse/dbplyr#1409]
# Overview

Getting an error using the new `dbplyr` and `bigrquery`. Had to revert back to `dbplyr 2.3.4` for it to work.

# Problem 1

```
library(dplyr)

conn <- DBI::dbConnect(
  bigrquery::bigquery(),
  project = "isb-cgc-bq",
  dataset = "TCGA_versioned"
)

conn |>
  DBI::dbListTables() |>
  head(1)

conn |> 
  tbl("DNA_methylation_chr10_hg19_gdc_2017_01")
```

See the error below...

```
Error in strsplit(name, ".", fixed = TRUE) : non-character argument
```

I dug into this a little and seems like `vctrs::field(x, 'schema')` might not be working correctly within `dbplyr:::check_from_for_query_or_schema` 

# Problem 2
Also doesn't work if I do 

```
conn |>
+   tbl(sql("select * from isb-cgc-bq.TCGA_versioned.clinical_gdc_r24"))
Error in curl::curl_fetch_memory(url, handle = handle) : 
  URL using bad/illegal format or missing URL
```

This one seems to break in `dplyr::make_tbl` .... specifically `structure(list(...), class = c(subclass, "tbl"))` ... the `tbl_sql` subclass seems to having some issues here, unclear whats going on. 

# Remarks

These errors are occur with code that previously worked using internal company schemas as well. The TCGA code is just for reproducibility. I might be wrong about my debugging but hopefully it can help. 

# Session Info:

```
> sessionInfo()
R version 4.3.2 (2023-10-31)
Platform: x86_64-apple-darwin20 (64-bit)
Running under: macOS Ventura 13.4.1

Matrix products: default
BLAS:   /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: America/Chicago
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] dplyr_1.1.3

loaded via a namespace (and not attached):
 [1] vctrs_0.6.4         httr_1.4.7          cli_3.6.1           rlang_1.1.2         DBI_1.1.3           bigrquery_1.4.2     purrr_1.0.2         generics_0.1.3      assertthat_0.2.1    jsonlite_1.8.7      bit_4.0.5           glue_1.6.2         
[13] gargle_1.5.2        openssl_2.1.1       dbplyr_2.4.0        askpass_1.2.0       fansi_1.0.5         rappdirs_0.3.3      cellranger_1.1.0    tibble_3.2.1        lifecycle_1.0.4     googlesheets4_1.1.1 compiler_4.3.2      fs_1.6.3           
[25] Rcpp_1.0.11         pkgconfig_2.0.3     rstudioapi_0.15.0   R6_2.5.1            tidyselect_1.2.0    utf8_1.2.4          pillar_1.9.0        curl_5.1.0          magrittr_2.0.3      bit64_4.0.5         tools_4.3.2         googledrive_2.1.1  
```

**Comment by HBuenoo** [comment]
Same issues here, while using dbplyr 2.4.0 even when getting a simple tbl(con, "tbl_name"), when con is from bigrquery, can't do it. Also solved when downgrading  to dbplyr 2.3.4

**Comment by hadley** [comment]
These should all be fixed in dev bigrquery, which I plan to release next week.
